### PR TITLE
Remove unused local variables project wide

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
- Removed unused or immediately overwritten variables
- Removed unused or immediately overwritten variables in tests
- Commented out some reference code, as it is not used in that scope
- Not sure on one prospect removal. Added a TODO.

### Testing Instructions

None, should not change behavior

### Documentation Changes Required

None.

### CODE REVIEW

File: libraries/idna_convert/idna_convert.class.php [hunk 1]
```diff
@@ -599,7 +597,6 @@ protected function _error($error = '')
     protected function _nameprep($input)
     {
         $output = array();
-        $error = false;
```
Comment (andrepereiradasilva) [inline libraries/idna_convert/idna_convert.class.php]: this is a vendor library i think. please revert the changes in this file.

Comment (mbabker) [inline libraries/idna_convert/idna_convert.class.php]: Yes, IDNA_Convert is a third party library.

Comment (frankmayer) [inline libraries/idna_convert/idna_convert.class.php]: Sorry, will revert.

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on be41cd69ee893d4b30f2567aa9e32013b1494722<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13206">issues.joomla.org/tracker/joomla-cms/13206</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on be41cd69ee893d4b30f2567aa9e32013b1494722<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13206">issues.joomla.org/tracker/joomla-cms/13206</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13206">issues.joomla.org/tracker/joomla-cms/13206</a>.</sub>
